Allow dotted instance names in `limactl start`

jandubois opened the pull request:
`limatmpl.Read()` no longer returns an empty template when called with an instance name because it now needs to be able to read any kind of file via the `template://` mechanism.

This commit changes the callers to no longer call `limatmpl.Read()` with an instance name. This also allows simplification of the locator type selection: every locator that is not a URL can now be treated as a local file.

Fixes #3574

AkihiroSuda commented on cmd/limactl/start.go:
```diff
@@ -148,9 +148,16 @@ func loadOrCreateInstance(cmd *cobra.Command, args []string, createOnly bool) (*
+	var tmpl *limatmpl.Template
+	// Only filenames ending in .yml or .yaml are template filenames; otherwise they are instance names.
+	// This check is necessary because limatmpl.Read() can also be used to read non-template files.
+	if limatmpl.SeemsFilePath(arg) && !(strings.HasSuffix(arg, ".yml") || strings.HasSuffix(arg, ".yaml")) {
```
Should be case-insensitive?

AkihiroSuda replied:
With unit tests, maybe

jandubois committed:
Allow dotted instance names in `limactl start`

limatmpl.Read() no longer returns an empty template when called with
an instance name because it now needs to be able to read any kind of
file via the template:// mechanism.

This commit changes the callers to no longer call limatmpl.Read() with
an instance name. This also allows simplification of the locator type
selection: every locator that is not a URL can now be treated as a local
file.

Signed-off-by: Jan Dubois <jan.dubois@suse.com>

jandubois commented:
@AkihiroSuda PTAL

AkihiroSuda reviewed (approved):
Thanks
